buildkite/resource_pipeline.go

```diff
@@ -153,11 +158,27 @@ func CreatePipeline(ctx context.Context, d *schema.ResourceData, m interface{})
 		return diag.FromErr(err)
 	}
 
+	teamPipelines := getTeamPipelinesFromSchema(d)
 	var mutation struct {
 		PipelineCreate struct {
 			Pipeline PipelineNode
-		} `graphql:"pipelineCreate(input: {cancelIntermediateBuilds: $cancel_intermediate_builds, cancelIntermediateBuildsBranchFilter: $cancel_intermediate_builds_branch_filter, defaultBranch: $default_branch, description: $desc, name: $name, organizationId: $org, repository: {url: $repository_url}, skipIntermediateBuilds: $skip_intermediate_builds, skipIntermediateBuildsBranchFilter: $skip_intermediate_builds_branch_filter, steps: {yaml: $steps}})"`
+		} `graphql:"pipelineCreate(input: {cancelIntermediateBuilds: $cancel_intermediate_builds, cancelIntermediateBuildsBranchFilter: $cancel_intermediate_builds_branch_filter, defaultBranch: $default_branch, description: $desc, name: $name, organizationId: $org, repository: {url: $repository_url}, skipIntermediateBuilds: $skip_intermediate_builds, skipIntermediateBuildsBranchFilter: $skip_intermediate_builds_branch_filter, steps: {yaml: $steps}, teams: $teams})"`
+	}
+
+	teamsData := make([]PipelineTeamAssignmentInput, 0)
+	for _, team := range teamPipelines {
+		log.Printf("converting team slug '%s' to an ID", string(team.Team.Slug))
+		teamID, err := GetTeamID(string(team.Team.Slug), client)
```

Comment:
this is a bit gross - for each team slug listed in the pipeline we do a graphql query to convert it into an ID. It's no worse than what we have on main though - `reconcileTeamPipelines()` also calls `GetTeamID()`